scipy/optimize/minpack.py

```diff
@@ -35,6 +35,7 @@ def _check_func(checker, argname, thefunc, x0, args, numinputs,
                 msg += " '%s'." % func_name
             else:
                 msg += "."
+            msg += 'Shape should be %s but it is %s.'.format(output_shape, shape(res))
```

Comment:
If you use `.format()`, you have to use `{}` as substitution character, here, you'll basically get `%s` in the interpolated string and no actual substitution. Otherwise use `%` instead of `.format()`.